_posts/en/newsletters/2020-06-24-newsletter.md

```diff
@@ -0,0 +1,254 @@
+endcomment %}
+{% assign bse = "https://bitcoin.stackexchange.com/a/" %}
+
+- [Why was the current formula that calculates 'target' from 'nBits' chosen?]({{bse}}96298)
+  Ravi Patel asks why a simpler formula for calculating the difficulty target
+  from `nBits` wasn’t chosen. Andrew Chow dives into some of the details around
+  the formula, its history, and even sample code from Bitcoin’s 0.1.5 release.
+
+- [Does Bitcoin really need timestamps?]({{bse}}96185)
+  Pieter Wuille explains why limiting the block rate without reference to a
+  clock time outside of the blockchain could make running full nodes more
+  expensive while also struggling to keep the stale block rate low and
+  preventing collusion attacks.
+
+- [In a fee overpayment attack, why can't compromised software provide fake previous transactions corresponding to fake inputs?]({{bse}}96309)
+  Regarding a [fee overpayment attack][news101 fee overpayment attack] on segwit
+  transactions with multiple inputs, justinmoon asks why the remedy of the
+  attack, requiring copies of previous transactions for the inputs, is not
+  vulnerable to malicious software providing fake previous transactions. Since
+  any provided previous transaction must have a hash that matches the spending input’s
+  previous transaction hash, such an attack is not feasible.
+
+## Releases and release candidates
+
+*New releases and release candidates for popular Bitcoin infrastructure
+projects.  Please consider upgrading to new releases or helping to test
+release candidates.*
+
+- [LND 0.10.2-beta.rc2][lnd 0.10.2-beta] this release candidate for an
+  LND maintenance release is now available for testing.
+
+## Notable code and documentation changes
+
+*Notable changes this week in [Bitcoin Core][bitcoin core repo],
+[C-Lightning][c-lightning repo], [Eclair][eclair repo], [LND][lnd repo],
+[Rust-Lightning][rust-lightning repo], [libsecp256k1][libsecp256k1 repo],
+[Bitcoin Improvement Proposals (BIPs)][bips repo], and [Lightning
+BOLTs][bolts repo].*
+
+- [Bitcoin Core #19260][] disconnects peers that send the [BIP37][]
+  `filterclear` message if the local node isn't accepting bloom filters
+  (as advertised using the [BIP111][] `NODE_BLOOM` service flag).  It
+  was [previously proposed][Bitcoin Core #8709] that nodes starting in
+  Initial Blockchain Download (IBD) could register as non-relaying peers
+  to avoid receiving recent transactions while they were still
+  downloading large numbers of blocks.  When they finished syncing, they
+  could then transition to receiving relayed transactions by sending a
+  `filterclear` message.  However, it was more recently
+  [proposed][Bitcoin Core #19204] that this could be done instead using
+  the [BIP133][] `feefilter` message.  That eliminates any need for a
+  non-bloom node to support the `filterclear` message and so this PR
+  removes that feature.
+
+- [Bitcoin Core #19133][] adds a bitcoin-cli `-generate` parameter (note leading dash) to replace
```

Comment:
IMO, `bitcoin-cli` is a command and `-generate` is a parameter or argument.

Reply:
They are commands in the help.
```
$ bitcoin-cli -help
Bitcoin Core RPC client version ...

Usage:  bitcoin-cli [options] <command> [params]  Send command to Bitcoin Core
or:     bitcoin-cli [options] -named <command> [name=value]...  Send command to Bitcoin Core (with named arguments)
or:     bitcoin-cli [options] help                List commands
or:     bitcoin-cli [options] help <command>      Get help for a command
```

In practice, of the commands and options listed afterwards in the help (in `src/bitcoin-cli.cpp`), most are options and two of them or more, `-getinfo` and `-generate` (and arguably the help/h/? help commands and `-version`), are commands.

Reply:
Nope, they're arguments in the help.  :-)  See this line specifically:

    or:     bitcoin-cli [options] -named <command> [name=value]...  Send command to Bitcoin Core (with named arguments)

What it's calling commands are RPCs.

Reply:
I see your point, but describing -generate as a bitcoin-cli param/arg, that itself takes params/args to pass to the RPC command, seems convoluted to me. Maybe "option"? The help lists them as "Options".

Perhaps prefixing "command" with "RPC" in the help would also be clearer. :man_shrugging:

Reply:
I've been reading that help incorrectly for years now :p

Reply:
I use "command" for the name of the program run on the *command* line.  I use "parameter" for attributes passed to a command or RPC that modify its behavior.  I try to avoid "argument" because it sometimes requires extra words to make it clear I'm not talking about an argument between people.  I try to avoid "option" unless the parameter is actually optional (which, in `bitcoin-cli`'s case, is currently all parameters).

Reply:
Thanks @harding, as always learned much from discussing with you. (I'll probably continue to think about it as a command running inside the cli program, but I see the cli help with new eyes now :)